src/main/java/io/fabric8/maven/docker/StopMojo.java

```diff
@@ -71,6 +72,41 @@
     @Parameter(property = "docker.stopNamePattern")
     private String stopNamePattern;
 
+    /**
+     * If true, the containers are not stopped right away, but when the build is finished (success or failed).
+     */
+    @Parameter(property = "docker.executeStopOnVMShutdown", defaultValue = "false")
+    private boolean executeStopOnVMShutdown;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        if (this.executeStopOnVMShutdown) {
+            this.executeStopOnVMShutdown = false;
+           if (!invokedTogetherWithDockerStart()) {
+                throw new MojoExecutionException("docker:stop with executeStopOnVMShutdown is only possible if" +
+                        " the docker containers started within the same maven session.");
+            }
+
+            try {
+                // HttpDelete class is not used during start mojo, so we need to load it and initialize it. It is not
+                // possible to load classes in the shutdown hook as
+                Class.forName("org.apache.http.client.methods.HttpDelete", true, this.getClass().getClassLoader());
+            } catch (ClassNotFoundException e) {
+                e.printStackTrace();
```

Comment:
Sonar reports this as a security hotspot: https://sonarcloud.io/project/security_hotspots?id=fabric8io_docker-maven-plugin&pullRequest=1492&hotspots=AXv_23_LXkHjdTy7bc6z 

Do you think we can avoid this by just logging some message instead of printing whole stacktrace?